### PR TITLE
Vladimir review

### PR DESCRIPTION
**Pull request by VladimirLi**

### CODE REVIEW

File: README.md
Comment (VladimirLi) [inline README.md:1]: 👍

File: src/assets/data.json
Comment (VladimirLi) [inline src/assets/data.json:1]: Somehow unrelated file?

File: src/App.tsx
Comment (VladimirLi) [inline src/App.tsx:1]: Somehow the formatting of the files is a bit off.
Perhaps due to the use of tabs instead of spaces?

File: src/App.tsx [hunk 1]
```diff
@@ -1,26 +1,30 @@
...
+				<Route
+					path="/"
+					element={<PokemonList />}
+				/>
+				<Route
+					path="pokemondetail"
+					element={
+						<PokemonDetailComponent />
+					}
+				/>
```
Comment (VladimirLi) [inline src/App.tsx:24]: Could have used dynamic routing

File: src/components/ListItemCard.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	const navigateToDetail = (
+		selectedName: string,
+		selectedUrl: string
+	) => {
+		pokemonContext?.loadDetail(
+			selectedUrl
+		);
+		pokemonContext?.setSelectedPokemon(
+			selectedName
+		);
```
Comment (VladimirLi) [inline src/components/ListItemCard.tsx:38]: This wouldn't have been necessary if dynamic routing were used

File: src/context/PokemonContext.tsx
Comment (VladimirLi) [inline src/context/PokemonContext.tsx:1]: Using context for data fetching 👍

File: src/pages/PokemonDetail.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+		setPokemonDetail,
+	] = useState<PokemonDetail | null>();
+
+	useEffect(() => {
+		setPokemonDetail(
+			pokemonContext?.pokemonDetail
+		);
+	}, [pokemonContext?.pokemonDetail]);
+	return (
+		<div className="DetailScreen">
```
Comment (VladimirLi) [inline src/pages/PokemonDetail.tsx:43]: Few other options are to use styled components or just inline styling.
I am not a big fan of mixing different ways of applying CSS, unless there is a real need for that.

File: src/pages/PokemonList.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+					height: 200,
+					width: 200,
+				}}
+			/>
+			<div className="Sitename">
+				<Typography variant="h5">
+					Pokemon Explorer
+				</Typography>
+			</div>
+
```
Comment (VladimirLi) [inline src/pages/PokemonList.tsx:45]: Somehow it is weird to see a mix of MUI and core html components

**Review by VladimirLi — COMMENTED** [review]
* Weird formatting → hard to read code
* Mix of different ways of applying CSS
* Use of react context 👍 
* Not very well-structured code

File: src/pages/PokemonList.tsx (hunk 4 above)
Comment (SammaniAri) [reply src/pages/PokemonList.tsx:45]: Fixed

File: src/pages/PokemonDetail.tsx (hunk 3 above)
Comment (SammaniAri) [reply src/pages/PokemonDetail.tsx:43]: fixed

File: src/App.tsx (hunk 1 above)
Comment (SammaniAri) [reply src/App.tsx:24]: fixed

File: src/components/ListItemCard.tsx (hunk 2 above)
Comment (SammaniAri) [reply src/components/ListItemCard.tsx:38]: fixed
